glTF-BinExporter/RhinoDocGltfConverter.cs

```diff
@@ -165,8 +169,11 @@ public byte[] GetBinaryBuffer()
 
         int[] GetMaterials(RenderMaterial[] materials, RhinoObject rhinoObject)
         {
+            RhinoObject[] subObjects = rhinoObject.GetSubObjects();
             int[] materialIndices = new int[materials.Length];
 
+            Dictionary<Color4f, int> colorIndices = new Dictionary<Color4f, int>();
```

Comment:
Shouldn't this Dictionary be a class member? If several objects share a color there will still be duplicate items if I'm reading this correctly. You should also use a different key than a Color4f. Perhaps an integer using the System.Drawing.Color.ToArgb(). Floating point precision may cause duplicate materials being stored. I'm also not sure the Color4f Hash function will work well in this situation.

Reply:
It depends. Should objects still have their own materials? Or should colors be shared for the while document?

My original intent was to merge sub-object materials. But we can extend it in the spirit of compact file size.

Reply:
No, materials should be able to be shared. Especially simple materials with only the base color set. This was the original functionality when using the materialsMap to track them by Guid to index.